week-4/E-arrays-of-objects/exercise-3.js

```diff
@@ -53,33 +53,62 @@ var restaurants = [restaurant1, restaurant2, restaurant3];
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
-
+function getRestaurantName(restaurants) {
+  return restaurants.name;
+}
+function getRestAvailable(rest, numberOfPeople) {
+  return rest.totalSeats - rest.numberOfCustomers >= numberOfPeople;
+}
 
 var restaurantFinderApplication = {
-    applicationName: "Restaurant Finder",
-    applicationVersion: "1.0",
-    restaurants: restaurants,
-    findAvailableRestaurants: function(numberOfPeople) {
-        // Complete here
-    },
-    findRestaurantServingDish: function(dishName) {
-        // Complete here
-    },
-    countNumberOfRestaurantsInArea: function(area) {
-        // Complete here
-    }
+  applicationName: "Restaurant Finder",
+  applicationVersion: "1.0",
+  restaurants: restaurants,
+  findAvailableRestaurants: function(numberOfPeople) {
+    return restaurants.filter( function getRestAvailable(rest) {
+      return rest.totalSeats - rest.numberOfCustomers >= numberOfPeople;
+      })
+      .map(getRestaurantName);
+  },
```

Comment:
good, this will work, but you have getRestAvailable defined twice, and are only using one.  

There is also no need to name the function you are creating and passing in-line to filter on line 68.

More importantly, note that the `restaurants` variable referred to on line 68 is NOT the restaurants property of this object!  It is the global `restaurants` variable.   This is probably not what you meant to do.